tests/integration/test_replay_events.py

```diff
@@ -136,3 +136,34 @@ def test_replay_events_without_processing(mini_sentry, relay_chain):
 
     replay_event = envelope.items[0]
     assert replay_event.type == "replay_event"
+
+
+def test_replay_events_are_filtered(
+    mini_sentry,
+    relay_with_processing,
+    replay_events_consumer,
+    outcomes_consumer,
+):
+    relay = relay_with_processing()
+    project_config = mini_sentry.add_full_project_config(
+        42, extra={"config": {"features": ["organizations:session-replay"]}}
+    )
+    filter_settings = project_config["config"]["filterSettings"]
+    filter_settings["localhost"] = {"isEnabled": True}
+    outcomes_consumer = outcomes_consumer()
+
+    replay_events_consumer = replay_events_consumer(timeout=10)
+    replay = generate_replay_sdk_event()
+    replay["request"]["url"] = "http://localhost:1200"
+
+    relay.send_replay_event(42, replay)
+
+    replay_events_consumer.assert_empty()
+
+    outcome = outcomes_consumer.get_outcome(timeout=5)
```

Comment:
I'd switch the order, first wait for the outcome, then assert empty. `assert_empty` just does the best effort thing, wait a tiny bit then assert there are no messages -> if we're already waiting longer we can use that and get better results in the tests.

Reply:
+1, and then call call `.assert_empty()` on the `outcomes_consumer` as well, just for good measure.